internal/dependency_cacher.go

```diff
@@ -28,33 +28,98 @@ func NewDependencyCacher(downloader Downloader, logger scribe.Logger) Dependency
 	}
 }
 
-func (dc DependencyCacher) Cache(root string, deps []cargo.ConfigMetadataDependency) ([]cargo.ConfigMetadataDependency, error) {
+type ConfigBuilpackOrExtensionMetadataDependency interface {
```

Comment:
I think this interface should be unexported (i.e. `type configBuilpackOrExtensionMetadataDependency interface`)